Jetty 12 - Move `GzipDefaultServletTest` tests to jetty-core/jetty-server tests

joakime opened the pull request.

joakime committed:
Move GzipDefaultServletTest to jetty-core/jetty-server tests

gregw reviewed (approved):
LGTM pending clean build

joakime committed:
Cleanup and organization of GzipHandlerTest.

Names of test methods cleaned up.
Duplicate tests removed.
Assertions standardized.
